### PR TITLE
fix: improve code specification

### PR DESCRIPTION
**Pull request by sh1luo**
Signed-off-by: sh1luo <690898835@qq.com>

### CODE REVIEW

**Comment by sh1luo** [comment]
In this pr, I mainly did four things:

1. add comment for exported function, which the first letter is capital.
2. Remove unnecessary code, such as `break`, `else`, Inappropriate usage of strings package and double quotes escape problem, etc. 

Finally, get `A` score in [Go Report](https://goreportcard.com/).

3. Re-write README, making the directory structure clearer and optimize the content. Now directory structure like this:
image

4. add good badges, which makes casdoor README looks better, online demo: [DEMO](https://github.com/sh1luo/casdoor/tree/fix-improve-code-specification)

**Comment by hsluoyz** [comment]
@kocoler @Kininaru @WindSpiritSR plz review

File: README.md [hunk 1]
```diff
@@ -1,146 +1,131 @@
...
+##### Dev Mode
 
-- Get code via `go get`:
+Edit `conf/app.conf`, set `runmode=dev`. Firstly build front-end files:
 
-    ```shell
-    go get github.com/casbin/casdoor
-    ```
+```bash
+cd web/ && npm run start
```
Comment (WindSpiritSR) [inline README.md:57]: `go mod tidy` & `npm install` ~

File: README.md [hunk 2]
```diff
@@ -1,146 +1,131 @@
...
 
-  http://localhost:8000/
+##### Prod Mode
 
-## Run (Dev Environment)
+Edit `conf/app.conf`, set `runmode=prod`. Firstly build front-end files:
 
-- Run backend (in port 8000):
+```bash
+cd web/ && npm run build
```
Comment (WindSpiritSR) [inline README.md:73]: same as above ~

**Review by WindSpiritSR — CHANGES_REQUESTED** [review]

**Comment by sh1luo** [comment]
@WindSpiritSR ok, fixed.

**Comment by hsluoyz** [comment]
lgtm

**Pull request merged**